examples/todo-intermediate/src/controllers/todo-list.controller.ts

```diff
@@ -0,0 +1,67 @@
+import {repository, HasManyRepository} from '@loopback/repository';
+import {TodoListRepository} from '../repositories';
+import {TodoList, Todo} from '../models';
+import {requestBody, param, post, get, del, put, patch} from '@loopback/rest';
+
+export class TodoListController {
+  protected todoRepo: HasManyRepository<TodoList>;
+
+  constructor(
+    @repository(TodoListRepository) protected todoListRepo: TodoListRepository,
+  ) {}
+
+  @post('/todo-lists')
+  async createTodoList() {
+    const newTodoList = {createdAt: new Date()};
+    return await this.todoListRepo.create(newTodoList);
+  }
+
+  @get('/todo-lists/{id}')
+  async findTodoListById(@param.path.number('id') id: number) {
+    return await this.todoListRepo.findById(id);
+  }
+
+  @get('/todo-lists')
+  async findTodos() {
+    return await this.todoListRepo.find();
+  }
+
+  @put('/todo-lists/{id}')
+  async replaceTodo(
+    @param.path.number('id') id: number,
+    @requestBody() todoList: TodoList,
+  ) {
+    return await this.todoListRepo.replaceById(id, todoList);
+  }
+
+  @patch('/todo-lists/{id}')
+  async updateTodo(
+    @param.path.number('id') id: number,
+    @requestBody() todoList: TodoList,
+  ) {
+    return await this.todoListRepo.updateById(id, todoList);
+  }
+
+  @del('/todo-lists/{id}')
+  async deleteTodo(@param.path.number('id') id: number) {
+    return await this.todoListRepo.deleteById(id);
+  }
+
+  @post('/todo-lists/{id}/todos')
+  async createTodoForTodoList(
```

Comment:
Please move methods implementing REST API for the relation "TodoList hasMany Todo" into a new controller `TodoListTodosController` (feel free to pick a better name). See https://github.com/strongloop/loopback-next/pull/1500#discussion_r201331287

> in LB4, we recommend to create a new controller for each relation in order to keep the controller classes smaller. I.e. don't add order-related methods to `CustomerController`, but create a new `CustomerOrdersController` class for them.